packages/bitcoin/src/utils/bitcoin.utils.ts

```diff
@@ -103,33 +104,37 @@ export function decodeBitcoinTx(tx: string): ReturnType<typeof btc.RawTx.decode>
 export function getAddressFromOutScript(
   script: Uint8Array,
   bitcoinNetwork: BtcSignerNetwork
-): string {
+): BitcoinAddress {
   const outputScript = btc.OutScript.decode(script);
 
   switch (outputScript.type) {
     case 'pkh':
     case 'sh':
     case 'wpkh':
     case 'wsh':
-      return btc.Address(bitcoinNetwork).encode({
-        type: outputScript.type,
-        hash: outputScript.hash,
-      });
+      return createBitcoinAddress(
+        btc.Address(bitcoinNetwork).encode({
+          type: outputScript.type,
+          hash: outputScript.hash,
+        })
+      );
     case 'tr':
-      return btc.Address(bitcoinNetwork).encode({
-        type: outputScript.type,
-        pubkey: outputScript.pubkey,
-      });
+      return createBitcoinAddress(
+        btc.Address(bitcoinNetwork).encode({
+          type: outputScript.type,
+          pubkey: outputScript.pubkey,
+        })
+      );
     case 'ms':
-      return btc.p2ms(outputScript.m, outputScript.pubkeys).address ?? '';
+      return createBitcoinAddress(btc.p2ms(outputScript.m, outputScript.pubkeys).address ?? '');
     case 'pk':
-      return btc.p2pk(outputScript.pubkey, bitcoinNetwork).address ?? '';
+      return createBitcoinAddress(btc.p2pk(outputScript.pubkey, bitcoinNetwork).address ?? '');
     case 'unknown':
-      return 'unknown';
+      return createBitcoinAddress('unknown');
     case 'tr_ms':
     case 'tr_ns':
     default:
-      return '';
+      return createBitcoinAddress('');
```

Comment:
Doesn't this throw an error? This isn't a bitcoin address

Reply:
Empty string is an unexpected value though, and not a valid bitcoin address (that BitcoinAddress ought to throw an error if it's given).

Reply:
`createBitcoinAddress('')` should throw an error as there is a check for not empty in `isValidBitcoinAddress` so this seems like a bug

Reply:
createBitcoinAddress('unknown') should also throw an error here too